src/spec_tools/core.cljc

```diff
@@ -130,7 +130,11 @@
         (-options [_] options)
         (-encoder [_ spec _]
           (or (get spec encode-key)
-              (get encoders (parse/type-dispatch-value (:type spec)))
+              (let [e (get encoders (parse/type-dispatch-value (:type spec)))]
+                (when e
+                  (fn [this x]
+                    (binding [*transformer* nil]
+                      (e this x)))))
```

Comment:
A `when-let` would consolidate things a bit:
```
(when-let [e (get encoders (parse/type-dispatch-value (:type spec)))]
  #(binding [*transformer* nil] (e %1 %2)))
```

Reply:
I agree with the `when-let` approach, but I would rather use the `fn` construct because I wanted to keep the same semantics of the `coerce` function using `this` and `x` so the reader can have some references. I think `%1` and `%2` makes it harder to understand what they represent.

Reply:
Fair enough.

Reply:
Added.